hono/test/items.test.ts

```diff
@@ -1,16 +1,40 @@
-// import app from '../src/index';
-//
-// const MOCK_ENV = {
-//  DB: {
-//    prepare: () => {
-//      // Mocked D1
-//    },
-//  },
-// };
+import { execSync } from 'child_process'
+import type { UnstableDevWorker } from 'wrangler'
+import { unstable_dev } from 'wrangler'
+import { validItem } from './factory/items'
 
-describe('GET /items', () => {
-  it('should return no items', async () => {
-    // const res = await app.request('/items', {}, MOCK_ENV);
-    // expect(await res.json()).toEqual([])
+describe('Items Unit Test', () => {
+  let worker: UnstableDevWorker
+
+  beforeAll(async () => {
+    worker = await unstable_dev('./src/index.ts', {
+      env: 'test',
+      experimental: {
+        disableExperimentalWarning: true,
+      },
+      ip: '127.0.0.1',
+      persistTo: './test/util/tmp-d1-db',
+    })
+  })
+
+  afterAll(async () => {
+    await execSync(`npm run db-clean`)
+    await worker.stop()
+  })
+
+  test('GET /items: should return no items', async () => {
+    const res = await worker.fetch('/api/items')
+    expect(res.status).toBe(200)
+    expect(await res.json()).toEqual([])
+  })
+
+  test('POST /items: should create and return one item', async () => {
+    const res = await worker.fetch('/api/items', {
+      method: 'POST',
+      body: JSON.stringify(validItem()),
+      headers: { 'Content-Type': 'application/json' },
+    })
+    expect(res.status).toBe(200)
+    expect(await res.json()).toMatchObject([validItem()])
```

Comment:
I'd add a TODO here stating that this unit test isn't exhaustive. `toMatchObject()` only verifies that that valid item is a subset of the response body, so *technically* this unit test is not complete (doesn't check for the `id` field to be equal to 1).

Or just add an assertion that the `id` must be equal to 1 :)

Reply:
Assertion for `id` added!